mxcubeweb/core/components/workflow.py

```diff
@@ -32,7 +32,10 @@ def get_available_workflows(self):
         return {"workflows": workflows}
 
     def submit_parameters(self, params):
-        HWR.beamline.workflow.set_values_map(params)
+        if params["wfpath"] == "Gphl" or params["type"] == "GphlWorkflow":
+            HWR.beamline.gphl_workflow.set_values_map(params)
```

Comment:
@rhfogh 
IS this correct  : `HWR.beamline.gphl_workflow.set_values_map(params)`
I could not see any set_values_map function in /mxcubecore/mxcubecore/HardwareObjects/Gphl/GphlWorkflow.py

Reply:
Well spotted. 

It is probably not correct. I suspect I added that at the start when I was (deliberately) duplicating the workflow code, and that this function is not and never was called. Could you  check: is the POST route for gphl_workflow ever used? Should it be? Again,  I suspect not, so both the route and the submit_parameters function could be removed. Once the GPhL workflow is started it does its own communication between MXCuBE and the workflow application, and handles communication to the UI via signals.

Reply:
The way it is organised, the mxcubecore side sends the GphlJsonParametersNeeded signal and listens for the  GphlParameterReturn signal until the UI is finished. The GphlUpdateUiParameters signal is also sent from the mxcubecore side.

The GphlParameterReturn serves two functions. It comes with two parameters, 'instruction' and 'parameters'. 'parameters' is a keyword-value dictionary of parameters. 'instruction' tells what whould be done with the parameters. If instruction is PARAMETERS_READY or PARAMETERS_CANCELLED the UI 'session' is terminated, eitehr successfully or unsuccessfully, and core side takes it from there. If instruction is anything else, it is interpreted as the name of a parameter that has been updated, and the parameters dictionary is passed to the releant update function which generates a response. When that response is ready, it is sent back to the UI with a GphlUpdateUiParameters signal.

So, the UI side must listen for GphlJsonParametersNeeded and GphlUpdateUiParameters, and be able to send GphlParameterReturn.

Reply:
Understood.
Very good i think i should start testing for now.